endless/src/endless/EndlessUsbToolDlg.cpp

```diff
@@ -5131,6 +5134,36 @@ bool CEndlessUsbToolDlg::CopyFilesToESP(const CString &fromFolder, const CString
 	return retResult;
 }
 
+bool CEndlessUsbToolDlg::CreatePersistentStorageFileOnexFAT(const CString& drive)
+{
+    HANDLE tmpfile;
+    CString storage = drive + L"\\endless\\persistent.img";
+    char marker[] = "endless_live_storage_marker";
+    uint64_t bytes;
+    ULARGE_INTEGER free_bytes;
+    DWORD written;
+    BOOL ret;
+
+    if (!GetDiskFreeSpaceEx(drive, &free_bytes, NULL, NULL))
+        return false;
+
+    bytes = free_bytes.QuadPart;
+    tmpfile = CreateFile(storage, GENERIC_WRITE, FILE_SHARE_WRITE, NULL, CREATE_NEW, FILE_ATTRIBUTE_NORMAL, NULL);
+    if (tmpfile == INVALID_HANDLE_VALUE)
+        return false;
+
+    ret = WriteFile(tmpfile, marker, strlen(marker), &written, NULL);
+
+    safe_closehandle(tmpfile);
+
+    if (!ret) {
+        DeleteFile(storage);
+        uprintf("Failed to write marker to persistent storage file.");
```

Comment:
we should print the string form of GetLastError. there is a whole family of macros to make this a bit easier.